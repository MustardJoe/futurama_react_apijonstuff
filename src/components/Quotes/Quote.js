import React from 'react';
import PropTypes from 'prop-types';

function Quote({ quoteObj }) {
  return (
    <section>
      <img src={quoteObj.image} />
      <p>{quoteObj.character}</p>
      <p>{quoteObj.quote}</p>
    </section>
  );
}

Quote.propTypes = {
  quoteObj: PropTypes.shape({
    image: PropTypes.string.isRequired,
    character: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
  }).isRequired
};

export default Quote;
