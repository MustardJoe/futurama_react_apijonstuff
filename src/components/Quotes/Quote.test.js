import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component test', () => {
  it('renders Quote item', () => {
    let quote =  {
      character: 'Bender',
      quote: 'Bender is the greatest',
      image: 'bestBenderImage',
    };


    const wrapper = shallow(<Quote quoteObj={quote} />);
    expect(wrapper).toMatchSnapshot();
  });
});
