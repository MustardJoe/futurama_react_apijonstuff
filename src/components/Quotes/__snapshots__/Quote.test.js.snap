// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component test renders Quote item 1`] = `
<section>
  <img
    src="bestBenderImage"
  />
  <p>
    Bender
  </p>
  <p>
    Bender is the greatest
  </p>
</section>
`;
